k2/csrc/cuda/array.h

```diff
@@ -61,22 +61,10 @@ class Array1 {
   // TODO(haowen): why do we need this version?
   Array1() : dim_(0), byte_offset_(0), region_(nullptr) {}
 
-  Array1(int32_t dim, RegionPtr region, int32_t byte_offset)
-      : dim_(dim), region_(region), byte_offset_(byte_offset) {}
-
-  Array1(ContextPtr ctx, int32_t size, T elem) {
```

Comment:
@danpovey , I deleted this version of constructor, as it seems capturing `*this` in lambda in constructor is not valid. Instead, we can initialize data with `operator=`
```
Array1<int> array(context, size);
array = 3;
```